### PR TITLE
[CI] Jenkins on Windows builds

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
Added Windows build & unit/example tests on PyTorch/CPU.  The backend is built with MSBuild and MinGW is no longer required.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by VoVAllen** [comment]
Is jenkins workspace on windows reused by different commit? Since it's not docker, do we need to clean the directory or other environment we've used at the end of the process?
Also it's better to indicate which pytorch version is used in windows build.

**Comment by VoVAllen** [comment]
Thanks! Looks good to me.

File: Jenkinsfile [hunk 1]
```diff
@@ -5,10 +5,19 @@ def init_git_submodule() {
...
+}
+
 def setup() {
   init_git_submodule()
 }
 
+def setup_win64() {
+  init_git_submodule_win64()
+}
+
```
Comment (jermainewang) [inline Jenkinsfile]: We could remove these `setup()` functions and directly call `init_xxx`.

File: Jenkinsfile [hunk 2]
```diff
@@ -23,6 +32,26 @@ def build_dgl() {
...
+    msbuild dgl.sln
+    COPY Release\\dgl.dll .
+    """
+  }
+  dir ("python") {
+    bat "DEL /S /Q build *.egg-info dist"
+    bat "pip3 uninstall -y dgl || (call )"  // ignore error
+    bat "python setup.py install --user"
+  }
+}
```
Comment (jermainewang) [inline Jenkinsfile]: Let's put these in a script file under `tests/scripts`: `task_build_dgl.sh` and `task_build_dgl.bat`.

File: Jenkinsfile [hunk 3]
```diff
@@ -31,6 +60,14 @@ def pytorch_unit_test(dev) {
   }
 }
 
+def pytorch_unit_test_win64(dev) {
+  withEnv(["DGL_LIBRARY_PATH=${env.WORKSPACE}\\build", "PYTHONPATH=${env.WORKSPACE}\\python"]) {
+    bat "python -m nose -v --with-xunit tests"
+    bat "python -m nose -v --with-xunit tests\\pytorch"
+    bat "python -m nose -v --with-xunit tests\\graph_index"
+  }
+}
```
Comment (jermainewang) [inline Jenkinsfile]: Similar to above. How about having `task_unit_tests.sh` and `task_unit_test.bat`? They can accept device and backend as arguments.

File: Jenkinsfile [hunk 4]
```diff
@@ -117,6 +179,20 @@ pipeline {
...
+        stage("Pytorch CPU (Windows)") {
+          agent { label "windows" }
+          stages {
+            stage("TH CPU Win64 unittest") {
+              steps { pytorch_unit_test_win64("CPU") }
+            }
+	    stage("TH CPU Win64 example test") {
+	      steps { example_test_win64("CPU") }
+	    }
+          }
```
Comment (jermainewang) [inline Jenkinsfile]: There seem to have some indent problem with this code.

**Pull request merged**